appcues/src/main/java/com/appcues/ui/AppcuesViewModel.kt

```diff
@@ -56,9 +57,9 @@ internal class AppcuesViewModel(
 
     init {
         viewModelScope.launch {
-            experienceRenderer.stateFlow.collectLatest { result ->
+            experienceRenderer.collectState(renderContext) { result ->
```

Comment:
knowing the renderContext is important because of this utility method that collects only events for that specific renderContext

Reply:
yes, this seems the key change here - being able to listen to the state machine updates for the specific state machine that this VM is rendering, through the shared ExperienceRenderer reference.